src/Data/List/Properties.agda

```diff
@@ -740,6 +714,50 @@ map-concatMap f g xs = begin
   concatMap (map f ∘′ g) xs
     ∎
 
+------------------------------------------------------------------------
+-- catMaybes
+
+catMaybes-concatMap : catMaybes {A = A} ≗ concatMap fromMaybe
+catMaybes-concatMap [] = refl
+catMaybes-concatMap (mx ∷ xs)
+  with IH ← catMaybes-concatMap xs
+  with mx
+... | just x  = cong (x ∷_) IH
+... | nothing = IH
+
+length-catMaybes : ∀ xs → length (catMaybes {A = A} xs) ≤ length xs
+length-catMaybes [] = ≤-refl
+length-catMaybes (mx ∷ xs)
+  with IH ← length-catMaybes xs
+  with mx
+... | just _  = s≤s IH
+... | nothing = m≤n⇒m≤1+n IH
+
+catMaybes-++ : (xs ys : List (Maybe A)) →
+               catMaybes (xs ++ ys) ≡ catMaybes xs ++ catMaybes ys
+catMaybes-++ []        _  = refl
+catMaybes-++ (mx ∷ xs) ys
+  with IH ← catMaybes-++ xs ys
+  with mx
+... | just x  = cong (x ∷_) IH
+... | nothing = IH
+
+module _ (f : A → B) where
+
+  map-catMaybes : map f ∘ catMaybes ≗ catMaybes ∘ map (Maybe.map f)
+  map-catMaybes [] = refl
+  map-catMaybes (mx ∷ xs)
+    with IH ← map-catMaybes xs
+    with mx
+  ... | just x  = cong (f x ∷_) IH
+  ... | nothing = IH
+
+Any-catMaybes⁺ : ∀ {P : Pred A ℓ} {xs : List (Maybe A)}
+  → Any (MAny P) xs → Any P (catMaybes xs)
+Any-catMaybes⁺ {xs = nothing ∷ xs} (there x∈)       = Any-catMaybes⁺ x∈
+Any-catMaybes⁺ {xs = just x  ∷ xs} (here (just px)) = here px
+Any-catMaybes⁺ {xs = just x  ∷ xs} (there x∈)       = there $ Any-catMaybes⁺ x∈
```

Comment:
Why here rather than added after L150 in the original?

Reply:
And for the sake of juxtaposition, I'd swap L758 and L759, as well as suppressing `x` and `xs`:
```suggestion
Any-catMaybes⁺ : ∀ {P : Pred A ℓ} {xs : List (Maybe A)}
  → Any (MAny P) xs → Any P (catMaybes xs)
Any-catMaybes⁺ {xs = nothing ∷ _} (there x∈)       = Any-catMaybes⁺ x∈
Any-catMaybes⁺ {xs = just _  ∷ _} (there x∈)       = there $ Any-catMaybes⁺ x∈
Any-catMaybes⁺ {xs = just _  ∷ _} (here (just px)) = here px
```

Reply:
> Why here rather than added after L150 in the original?

Grouped the properties of `catMaybe` with the ones for `mapMaybe`, as it seemed more intuitive.
Also, lemma `catMaybes-concatMap` mentions `concatMap` which comes just before in the file.

Reply:
Well, perhaps there is no 'right' order for module contents, so the rule-of-thumb ends up being: try, where possible, to follow in `Properties` the order in `Base`... but even that seems hard to stick to.